### PR TITLE
Remove Travis build shield from README

### PR DESCRIPTION
**Pull request by mokagio**
We don't use Travis anymore, since #1091.

### Test
N.A

### Review
Only one developer required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Simplenote for Android
-[![Version](https://img.shields.io/badge/version-2.19-blue.svg)](https://github.com/Automattic/simplenote-android/releases/tag/2.19) [![CircleCI](https://img.shields.io/circleci/build/gh/Automattic/simplenote-android.svg?label=circleci)](https://circleci.com/gh/Automattic/simplenote-android) [![Travis CI](https://img.shields.io/travis/Automattic/simplenote-android/develop.svg?label=travisci)](https://travis-ci.org/Automattic/simplenote-android)
+[![Version](https://img.shields.io/badge/version-2.19-blue.svg)](https://github.com/Automattic/simplenote-android/releases/tag/2.19) [![CircleCI](https://img.shields.io/circleci/build/gh/Automattic/simplenote-android.svg?label=circleci)](https://circleci.com/gh/Automattic/simplenote-android)
```
Comment (mokagio) [inline README.md]: For what is worth, I'd say we should remove the version and the CircleCI shields, too. Those shields provide information that is already available on GitHub:

Screen Shot 2021-08-23 at 1 06 44 pm

Comment (AliSoftware) [reply README.md]: I agree, let's remove them all.

This information (CI status and latest version) was not as visible a while ago with previous GitHub UI, but now that it is, it's not really relevant anymore.

(In which case we also need to remove the step that bumps that version in the badge during release finalization btw)

**Review by AliSoftware — APPROVED** [review]
@mokagio How was the `Version` badge – that this PR removed – updated on every release before? Since that's a step we might want to remove as well

I've tried a quick search of the term `README` in the Fastfile and scripts and couldn't find anything that seem to automate the update of that badge during release steps, and couldn't find anything in the release scenario either to suggest to do that step manually…

So was this all just manual and implicit and never documented in the scenario, which means nothing to remove on that front either? (Approving as I'm assuming 👈 this is the case)

**Comment by mokagio** [comment]
> that's a step we might want to remove as well

100%. In fact, I did remove it while working on this, but never opened a PR in our private repo with the release process checklist 😅

**Pull request merged**